src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpBehavior.cs

```diff
@@ -0,0 +1,1356 @@
+        {
+            if (isStream)
+            {
+                return s_defaultStreamContentType;
+            }
+            else if (useJson)
+            {
+                return JsonMessageEncoderFactory.GetContentType(webEncoding);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        IDispatchMessageFormatter GetDefaultDispatchFormatter(OperationDescription od, bool useJson, bool isWrapped)
+        {
+            DataContractSerializerOperationBehavior dcsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<DataContractSerializerOperationBehavior>();
+            if (useJson)
+            {
+                if (dcsob == null)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.JsonFormatRequiresDataContract, od.Name, od.DeclaringContract.Name, od.DeclaringContract.Namespace)));
+                }
+                return CreateDataContractJsonSerializerOperationFormatter(od, dcsob, isWrapped);
+            }
+            else
+            {
+                EndpointDispatcher dummyED = new EndpointDispatcher(new EndpointAddress("http://localhost/"), "name", "");
+                DispatchRuntime dispatchRuntime = dummyED.DispatchRuntime;
+                DispatchOperation dop = new DispatchOperation(dispatchRuntime, "dummyDispatch", "urn:dummy");
+                dop.Formatter = null;
+
+                if (dcsob != null)
+                {
+                    (dcsob as IOperationBehavior).ApplyDispatchBehavior(od, dop);
+                    return dop.Formatter;
+                }
+
+                XmlSerializerOperationBehavior xsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<XmlSerializerOperationBehavior>();
+                if (xsob != null)
+                {
+                    xsob = new XmlSerializerOperationBehavior(od, xsob.XmlSerializerFormatAttribute, _contractNamespace);
+                    (xsob as IOperationBehavior).ApplyDispatchBehavior(od, dop);
+                    return dop.Formatter;
+                }
+            }
+
+            return null;
+        }
+
+        internal virtual DataContractJsonSerializerOperationFormatter CreateDataContractJsonSerializerOperationFormatter(OperationDescription od, DataContractSerializerOperationBehavior dcsob, bool isWrapped)
+        {
+            //return new DataContractJsonSerializerOperationFormatter(od, dcsob.MaxItemsInObjectGraph, dcsob.IgnoreExtensionDataObject, isWrapped, false, JavascriptCallbackParameterName);
+            return new DataContractJsonSerializerOperationFormatter(od, dcsob.MaxItemsInObjectGraph, dcsob.IgnoreExtensionDataObject, isWrapped, false, null);
+        }
+
+        // TODO: Can I remove this?
+        //IClientMessageFormatter GetDefaultXmlAndJsonClientFormatter(OperationDescription od, bool isWrapped)
+        //{
```

Comment:
Yes, you can remove this

Reply:
This was addressed in b54a2808ec6f4245ccf662c73a2b441ea6c85814.